### ISSUE TITLE
Support for running a WebGUI inside of Hass.io 

### ISSUE THREAD

**Issue by RubenKelevra** [ipfs-shipyard/ipfs-webui#1695]
HomeAssistant - an internet of things hub - has a dedicated operating system called Hass.io.

This operating system is lightweight and WebGUI based, which supports easy to install (one click) installation of add-ons which can are run in containers. As container system they use Docker.

Examples of stuff currently in the community repository:
- tor node 
- matrix server
- wireguard VPN

An easy to install IPFS node could in the future enable us to use the network to access the WebGUI of HomeAssistant like through a VPN. 

It would also allow us to easily put up lightweight nodes, power consumption wise, since those hubs run 24/7 anyway. 

Hope this is interesting :) 

https://www.home-assistant.io/addons/

**Comment by jessicaschilling** [comment]
Hi @RubenKelevra -- this is an interesting idea, thank you for bringing it up! Realistically speaking, it's not likely to be something that the core team can add to their workload any time in the near future, but it would be awesome as a community-led effort. Suggest bringing this up in https://discuss.ipfs.io to see if it gains any traction?

**Comment by RubenKelevra** [comment]
Hey thanks! I was thinking maybe it's worth to put out a small bounty for implementation and one for one year maintaining or similar.

The HomeAssistant community is growing quite rapidly and I'm sure there would be quite some interest if this could replace the tor solution or the payed webservice.

Additionally the network wouldn't gain some reliably nodes which runs 24/7 in ISP networks all around the globe and not in data centers.

I'm somewhat time limited in the next weeks, so maybe after that.

I would still like to just keep the ticket here (or moved somewhere else) and open to keep track of the progress :)

**Comment by autonome** [comment]
Would be interested in bountying this. @RubenKelevra if you or someone who knows the details from that HA community drafted a one-page spec for a minimal implementation, might be a good place to get started.

**Comment by jay-dee7** [comment]
This sounds so exciting. @jessicaschilling would it be possible for me to get involved here? I would love to contribute for this

**Comment by jessicaschilling** [comment]
@jay-dee7 - as noted above, we don't have a specific spec for this, but would rely on one being written up by @RubenKelevra or another community member. Maybe the two of you could collaborate on writing up that initial spec?

**Comment by jay-dee7** [comment]
Sure I'd love that. @RubenKelevra I'm open for collaboration for this if you have some time over the weekend for a quick zoom call. I haven't used HA before but from the docs it looks pretty straight forward.

**Comment by RubenKelevra** [comment]
> Sure I'd love that. @RubenKelevra I'm open for collaboration for this if you have some time over the weekend for a quick zoom call. I haven't used HA before but from the docs it looks pretty straight forward. 

Yeah sure we can look into that. Note that I'm just having the user perspective right now into HA. Never really looked under the hood.

But what I've read on docs this should be easily doable, good rights management and long term support to keep the two fast developing projects in working order for say a year would be the main focus I guess.

HA can create snapshots of itself and encrypt them with a password. This might be an interesting use case, that you can add the snapshots to ipfs and push it to a pinning service. We just need to make sure that the user understands that the ipfs databases won't be in any snapshot. 

That are my thoughts right now.

I'm more into a chat, easier for me as non-native English speaker. I'm at UTC+1, feel free to write me @RubenKelevra:matrix.org

**Comment by SgtPooki** [comment]
@RubenKelevra has there been any headway for this? This is very interesting. I'm a big fan of home automation, and would love to work on this when there's time

cc @tinytb.
